### PR TITLE
Add get tenants endpoint

### PR DESCRIPTION
**Pull request by dirkkul**
### What's being changed:
- Adds new endpoint to list all tenants of a class

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: usecases/schema/tenant.go [hunk 1]
```diff
@@ -238,3 +238,31 @@ func (m *Manager) onDeleteTenants(ctx context.Context, class *models.Class, req
 
 	return nil
 }
+
+// GetTenants is used to delete tenants of a class.
```
Comment (antas-marcin) [inline usecases/schema/tenant.go:242]: I think that it should be:
```
GetTenants is used to fetch tenants of a class.
```

File: usecases/schema/tenant.go [hunk 2]
```diff
@@ -238,3 +238,31 @@ func (m *Manager) onDeleteTenants(ctx context.Context, class *models.Class, req
...
+	if cls == nil {
+		return nil, fmt.Errorf("class %q: %w", class, ErrNotFound)
+	}
+	if !schema.MultiTenancyEnabled(cls) {
+		return nil, fmt.Errorf("multi-tenancy is not enabled for class %q", class)
+	}
+
+	var tenants []*models.Tenant
+	m.shardingStateLock.Lock()
+	if ss := m.state.ShardingState[cls.Class]; ss != nil {
```
Comment (antas-marcin) [inline usecases/schema/tenant.go:260]: Just wanted to be sure, if Weaviate is configured to run in multi-cluster setup `m.state.ShardingState` will return also proper list of Tenants?

Comment (dirkkul) [reply usecases/schema/tenant.go:260]: Tried a two node setup, added on node1 and queried on node2 and it worked

Comment (antas-marcin) [reply usecases/schema/tenant.go:260]: Good!

File: usecases/schema/tenant.go [hunk 3]
```diff
@@ -238,3 +238,32 @@ func (m *Manager) onDeleteTenants(ctx context.Context, class *models.Class, req
...
+	if !schema.MultiTenancyEnabled(cls) {
+		return nil, fmt.Errorf("multi-tenancy is not enabled for class %q", class)
+	}
+
+	m.shardingStateLock.Lock()
+	var tenants []*models.Tenant
+	if ss := m.state.ShardingState[cls.Class]; ss != nil {
+		tenants = make([]*models.Tenant, len(ss.Physical))
+		for tenant := range ss.Physical {
+			tenants = append(tenants, &models.Tenant{Name: tenant})
```
Comment (antas-marcin) [inline usecases/schema/tenant.go:263]: when you create a slice with a fixed size for example 4, then first 4 entries are empty, and you put those entries there with accessing the slice with index like `arrar[i] = value`, but when you use `append` it then adds more data to already created slice you will end up with 4+ lengith

Comment (dirkkul) [reply usecases/schema/tenant.go:263]: juuuust noticed

Comment (antas-marcin) [reply usecases/schema/tenant.go:263]: In this case you should create either create array with:
```
tenants := []*models.Tenant{}
```
or
```
tenants = make([]*models.Tenant, 0)
```

**Review by antas-marcin — APPROVED** [review]

**Pull request merged**